goal_src/levels/lavatube/lavatube-energy.gc

```diff
@@ -5,3 +5,1946 @@
+                                    )
+                                  (run-now-in-process gp-1 energyball-init-by-other (-> self root-override trans))
+                                  (-> gp-1 ppointer)
+                                  )
+                                )
+              )
+        )
+      (go energyarm-idle)
+      )
+    )
+  (none)
+  )
+
+(defskelgroup *energyhub-sg* energyhub
+  0
+  2
+  ((1 (meters 999999)))
+  :bounds (static-spherem 0 2 0 10)
+  :longest-edge (meters 0)
+  )
+
+(defbehavior energyhub-trans energyhub ()
+  (+!
+    (-> self y-rotation)
+    (* 36.40889
+       (-> self rotation-speed value)
+       (the float (- (-> *display* base-frame-counter) (-> *display* old-base-frame-counter)))
+       )
+    )
+  (cond
+    ((< 65536.0 (-> self y-rotation))
+     (set! (-> self y-rotation) (+ -65536.0 (-> self y-rotation)))
+     )
+    ((< (-> self y-rotation) 0.0)
+     (set! (-> self y-rotation) (+ 65536.0 (-> self y-rotation)))
+     )
+    )
+  (let ((gp-0 (new 'stack-no-clear 'matrix)))
+    (let* ((f0-9 (the float (-> *display* base-frame-counter)))
+           (f28-0 (- f0-9 (* (the float (the int (/ f0-9 1200.0))) 1200.0)))
+           )
+      (matrix-rotate-y! gp-0 (-> self y-rotation))
+      (matrix*! (-> self rot-mat) gp-0 (-> self rot-mat-init))
+      (set! (-> self x-rotation) (* 728.1778 (sin (* 54.613335 f28-0))))
+      )
+    (matrix-rotate-x! gp-0 (-> self x-rotation))
+    (matrix*! (-> self rot-mat) gp-0 (-> self rot-mat))
+    )
+  (matrix->quaternion (-> self root quat) (-> self rot-mat))
+  )
+
+(defbehavior energyhub-set-lava-height energyhub ()
+  (let ((v1-0 (-> self alts 2)))
+    (when v1-0
+      (let ((v1-2 (-> v1-0 extra process)))
+        (when v1-2
+          (let* ((f0-1 (- 0.0 (-> self palette-val)))
+                 (f0-2 (if (< 0.0 f0-1)
+                           (fmin 0.1 f0-1)
+                           (fmax -0.1 f0-1)
+                           )
+                       )
+                 )
+            (+! (-> self palette-val) f0-2)
+            )
+          (set! (-> *palette-fade-controls* control 2 fade) (-> self palette-val))
+          (set! (-> (the-as process-drawable v1-2) root scale x) 1.25)
+          (let ((f0-6 1.25))
+            (set! (-> (the-as process-drawable v1-2) root scale z) f0-6)
+            f0-6
+            )
+          )
+        )
+      )
+    )
+  )
+
+(defstate energyhub-stopped (energyhub)
+  :enter
+  (behavior ()
+    (set! (-> self rotation-speed target) 0.0)
+    (set! (-> self rotation-speed accel) 0.0005)
+    (dotimes (gp-0 5)
+      (send-event (handle->process (-> self arm gp-0)) 'stop)
+      )
+    (if (nonzero? (-> self sound))
+        (stop! (-> self sound))
+        )
+    (none)
+    )
+  :trans
+  (behavior ()
+    (update! (-> self rotation-speed) 0.0)
+    (energyhub-trans)
+    (let ((t9-2 energyhub-set-lava-height))
```

Comment:
I think this function takes a float argument

Reply:
ya you're right, but its unused -- weird.